frontend/src/concepts/pipelines/context/usePipelineEvents.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { groupVersionKind } from '#~/api/k8sUtils';
+import { EventKind, KnownLabels } from '#~/k8sTypes';
+import { CustomWatchK8sResult } from '#~/types';
+import { EventModel, PodModel } from '#~/api/models/k8s';
+import useK8sWatchResourceList from '#~/utilities/useK8sWatchResourceList';
+
+// ideally ,would use react-query watch many, but that is not available
+//react query supports this and we do not unfortunately
+// need to have one call per pod because react hooks need to be called the same amount of times for each rendering
+// TECH DEBT
+export const useWatchPipelineServerEvents = (
+  namespace: string,
+  podUid: string,
+): CustomWatchK8sResult<EventKind[]> =>
+  useK8sWatchResourceList(
+    {
+      isList: true,
+      groupVersionKind: groupVersionKind(EventModel),
+      namespace,
+      fieldSelector: `involvedObject.kind=Pod,involvedObject.uid=${podUid}`,
+    },
+    EventModel,
+  );
+
+export const useWatchPodsForPipelineServerEvents = (
+  namespace: string,
+): CustomWatchK8sResult<EventKind[]> =>
+  useK8sWatchResourceList(
+    {
+      isList: true,
+      groupVersionKind: groupVersionKind(PodModel),
+      namespace,
+      selector: { matchLabels: { component: KnownLabels.LABEL_SELECTOR_DATA_SCIENCE_PIPELINES } },
+    },
+    PodModel,
+  );
+
+// Custom hook to watch events for multiple pods
+export const useWatchMultiplePodEvents = (namespace: string, podUids: string[]): EventKind[] => {
+  // Always call hooks to maintain consistent order, but use empty string for non-existent pods
+  // This will result in empty event arrays for non-existent pods
+  const pod0Events = useWatchPipelineServerEvents(namespace, podUids[0] ?? '');
+  const pod1Events = useWatchPipelineServerEvents(namespace, podUids[1] ?? '');
+  const pod2Events = useWatchPipelineServerEvents(namespace, podUids[2] ?? '');
+  const pod3Events = useWatchPipelineServerEvents(namespace, podUids[3] ?? '');
+  const pod4Events = useWatchPipelineServerEvents(namespace, podUids[4] ?? '');
+  const pod5Events = useWatchPipelineServerEvents(namespace, podUids[5] ?? '');
+  const pod6Events = useWatchPipelineServerEvents(namespace, podUids[6] ?? '');
+  const pod7Events = useWatchPipelineServerEvents(namespace, podUids[7] ?? '');
+  const pod8Events = useWatchPipelineServerEvents(namespace, podUids[8] ?? '');
+  const pod9Events = useWatchPipelineServerEvents(namespace, podUids[9] ?? '');
```

Comment:
dont do this. there might not always be the same number of pipeline server related pods. make it dynamic instead